common/djangoapps/terrain/stubs/http.py

```diff
@@ -217,6 +218,8 @@ def _format_msg(self, format_str, *args):
         `format_str` is a string with old-style Python format escaping;
         `args` is an array of values to fill into the string.
         """
+        if not args:
+            format_str = urllib.unquote(format_str)
```

Comment:
@muzaffaryousaf   Something still doesn't feel right about this.  
* Are you saying that `format_str` is really always `url_format_str` and should be renamed?
* Why is this only being done `if not args`?
* It seems possible to me that this unquoting should actually happen at an earlier point in time where you know what you have.  It's sort of like the "unicode sandwich" concept ([search for it in this page](http://nedbatchelder.com/text/unipain.html)).  Things should be escaped/unescaped at the ends, and should be regular strings in the middle.